Sources/EngineKit/HashableExtensions.swift

```diff
@@ -46,118 +52,166 @@ extension Instructions: Hashable {
 // MARK: - Instruction + Hashable
 extension Instruction: Hashable {
 	public func hash(into hasher: inout Hasher) {
+		func mustNotHaveAssociatedValue(_: Instruction) {}
```

Comment:
What is the point of this?

Reply:
Is the idea to cause a compilation error if someone later adds an associated value to one of those cases? If so, what is the benefit of having that check here, and not where the enum is defined?

Reply:
@kugel3 where do you wanna have that check?

Reply:
I think it also deserves a comment, like "if an associated value is added to one of the cases called below, then it will no longer be possible to create the case and the corresponding line will fail to compile"

Reply:
sure! I thought the name was clear enough....

Reply:
Except that the name is a lie, you can call that function with any instruction, also ones that do have an associated value. It's not the function that enforces it, it's the call site.

Reply:
resolved on slack